examples/complete/main.tf

```diff
@@ -1,3 +1,120 @@
+
+  resources {
+    service              = "cloudantnosqldb"
+    resource_instance_id = element(split(":", module.create_cloudant.instance_guid), 7)
+  }
+}
+
+##############################################################################
+## Create ServiceId with access to Cloudant instance, via the access group
+## created above
+##############################################################################
+
+resource "ibm_iam_service_id" "cloudant_manager" {
+  name        = "${var.prefix}-cloudant-manager"
+  description = "ServiceID that has got manager and operator roles on the Cloudant instance"
+}
+
+resource "ibm_iam_access_group_members" "accgroupmem" {
+  access_group_id = ibm_iam_access_group.access_group.id
+  iam_service_ids = [ibm_iam_service_id.cloudant_manager.id]
+}
+
+########################################
+## Create Secrets Manager layer
+########################################
+
+# Create Secrets Manager Instance
+resource "ibm_resource_instance" "secrets_manager" {
+  count             = var.existing_sm_instance_guid == null ? 1 : 0
+  name              = "${var.prefix}-sm" #checkov:skip=CKV_SECRET_6: does not require high entropy string as is static value
+  service           = "secrets-manager"
+  service_endpoints = "public-and-private"
+  plan              = "trial"
+  location          = local.sm_region
+  resource_group_id = module.resource_group.resource_group_id
+  tags              = var.resource_tags
+
+  timeouts {
+    create = "30m" # Extending provisioning time to 30 minutes
+  }
+}
+
+# Add a Secrets Group to the secret manager instance
+module "secrets_manager_group" {
+  source                   = "terraform-ibm-modules/secrets-manager-secret-group/ibm"
+  version                  = "1.1.0"
+  region                   = local.sm_region
+  secrets_manager_guid     = local.sm_guid
+  secret_group_name        = "${var.prefix}-cloudant-secrets"
+  secret_group_description = "service secret-group"
+}
+
+##################################################################
+## Configure Secret Manager to generate and rotate API Key for the Service ID created above
+##################################################################
+
+#module "dynamic_cloudant_serviceid_apikey" {
+#  source                               = "git::https://github.ibm.com/GoldenEye/iam-serviceid-apikey-secrets-manager-module?ref=2.0.1"
+#  region                               = local.sm_region
+#  sm_iam_secret_name                   = "${var.prefix}-cloudant"
+#  sm_iam_secret_description            = "Dynamic IAM secret / apikey"
+#  serviceid_id                         = ibm_iam_service_id.cloudant_manager.id
+#  secrets_manager_guid                 = local.sm_guid
+#  secret_group_id                      = module.secrets_manager_group.secret_group_id
+#  sm_iam_secret_auto_rotation          = true
+#  sm_iam_secret_auto_rotation_interval = 85
+#  sm_iam_secret_auto_rotation_unit     = "day"
+#}
```

Comment:
Leave a comment to say this code block will be updated when module it moved to public TIM org. Also please make sure we have a git issue to track that